scheduler/data-flow/src/main/kotlin/io/seldon/dataflow/kafka/PipelineStep.kt

```diff
@@ -97,63 +103,53 @@ fun stepFor(
+    data class Many(val topicNames: Set<TopicForPipeline>) : SourceProjection()
+    data class ManySubsets(val tensorsByTopic: Map<TopicForPipeline, Set<TensorName>>) : SourceProjection()
 }
 
-fun parseTriggers(sources: List<TopicName>): Map<TopicName,Set<TensorName>> {
+fun parseTriggers(sources: List<PipelineTopic>): Map<TopicForPipeline,Set<TensorName>> {
     return sources
         .map { parseSource(it) }
-        .groupBy(keySelector = { it.first }, valueTransform = { it.second })
+        .groupBy(keySelector = { it.first+":"+it.third }, valueTransform = { it.second })
         .mapValues { it.value.filterNotNull().toSet() }
-        .map { TopicTensors(it.key, it.value) }
-        .associate { it.topicName to it.tensors }
+        .map { TopicTensors(TopicForPipeline(topicName = it.key.split(":")[0], pipelineName = it.key.split(":")[1]), it.value) }
+        .associate {it.topicForPipeline to it.tensors }
 }
 
-fun parseSources(sources: List<TopicName>): SourceProjection {
+fun parseSources(sources: List<PipelineTopic>): SourceProjection {
     val topicsAndTensors = sources
         .map { parseSource(it) }
-        .groupBy(keySelector = { it.first }, valueTransform = { it.second })
+        .groupBy(keySelector = { it.first+":"+it.third }, valueTransform = { it.second })
         .mapValues { it.value.filterNotNull().toSet() }
-        .map { TopicTensors(it.key, it.value) }
+        .map { TopicTensors(TopicForPipeline(topicName = it.key.split(":")[0], pipelineName = it.key.split(":")[1]), it.value) }
 
     return when {
         topicsAndTensors.isEmpty() -> SourceProjection.Empty
         topicsAndTensors.size == 1 && topicsAndTensors.first().tensors.isEmpty() ->
-            SourceProjection.Single(topicsAndTensors.first().topicName)
+            SourceProjection.Single(topicsAndTensors.first().topicForPipeline)
         topicsAndTensors.size == 1 ->
             SourceProjection.SingleSubset(
-                topicsAndTensors.first().topicName,
+                topicsAndTensors.first().topicForPipeline,
                 topicsAndTensors.first().tensors,
             )
         topicsAndTensors.all { it.tensors.isEmpty() } ->
-            SourceProjection.Many(topicsAndTensors.map { it.topicName }.toSet())
+            SourceProjection.Many(topicsAndTensors.map { it.topicForPipeline }.toSet())
         else ->
             SourceProjection.ManySubsets(
-                topicsAndTensors.associate { it.topicName to it.tensors },
+                topicsAndTensors.associate { it.topicForPipeline to it.tensors },
             )
     }
 }
 
-fun parseSource(source: TopicName): Pair<TopicName, TensorName?> {
-    return if (source.split(".").size > 5) {
-        source.substringBeforeLast(".") to source.substringAfterLast(".", "")
+fun parseSource(source: PipelineTopic): Triple<TopicName, TensorName?, String> {
+    return if (source.topicName.split(".").size > 5 ){
```

Comment:
why > 5, should we be more exact and do equality? (I understand that this is not something that this PR has introduced)